components/ml/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java

```diff
@@ -2637,4 +2637,35 @@ public void updateSamplePoints(long datasetVersionId, SamplePoints samplePoints)
             MLDatabaseUtils.closeDatabaseResources(connection, updateStatement);
         }
     }
+    
+    @Override
+    public boolean isValidModelStatus(long modelId, int tenantId, String userName) throws DatabaseHandlerException {
+        Connection connection = null;
+        ResultSet result = null;
+        PreparedStatement statement = null;
+        try {
+            connection = dbh.getDataSource().getConnection();
+            statement = connection.prepareStatement(SQLQueries.GET_MODEL_STATUS);
+            statement.setLong(1, modelId);
+            statement.setInt(2, tenantId);
+            statement.setString(3, userName);
+            result = statement.executeQuery();
+            if (result.first()) {
+                if (result.getString(1).equalsIgnoreCase(MLConstants.MODEL_STATUS_COMPLETE)) {
+                    return true;
+                }
+            } else {
+                throw new DatabaseHandlerException("Failed to find the model for model id " + modelId);
+            }
+        } catch (SQLException e) {
+            throw new DatabaseHandlerException(
+                    "An error has occurred while fetching the status of the model for model id: " + modelId + ": "
+                            + e.getMessage(), e);
+        } finally {
+            // Close the database resources.
+            MLDatabaseUtils.closeDatabaseResources(connection, statement, result);
+        }
+        // consider anything other than "Complete" status as an invalid model.
```

Comment:
Use proper linguistics for comments. Use capital letters to begin the sentences with.